src/watchdog_fsevents.c

```diff
@@ -798,9 +831,14 @@ static PyObject *
 watchdog_stop(PyObject *self, PyObject *emitter_thread)
 {
     UNUSED(self);
-    PyObject *value = PyDict_GetItem(thread_to_run_loop, emitter_thread);
-    if (G_IS_NULL(value)) {
-      goto success;
+    PyObject *value = NULL;
+    switch (PyDict_GetItemRef(thread_to_run_loop, emitter_thread, &value)){
+        case -1:
+            return NULL;
+        case 0:
+            goto success;
+        default:
+            break;
```

Comment:
I can't comment below but if the `G_RETURN_NULL_IF` macro includes a `return` statement then you also need to `DECREF` `value` if that happens.